Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -6,69 +6,147 @@
 
 .. _chaining_aws_cni:
 
-*******
-AWS-CNI
-*******
-
-This guide explains how to set up Cilium in combination with aws-cni. In this
-hybrid mode, the aws-cni plugin is responsible for setting up the virtual
-network devices as well as address allocation (IPAM) via ENI. After the initial
-networking is setup, the Cilium CNI plugin is called to attach eBPF programs to
-the network devices set up by aws-cni to enforce network policies, perform
-load-balancing, and encryption.
+******************
+AWS VPC CNI plugin
+******************
+
+This guide explains how to set up Cilium in combination with the AWS VPC CNI
+plugin. In this hybrid mode, the AWS VPC CNI plugin is responsible for setting
+up the virtual network devices as well as for IP address management (IPAM) via
+ENIs. After the initial networking is setup for a given pod, the Cilium CNI
+plugin is called to attach eBPF programs to the network devices set up by the
+AWS VPC CNI plugin in order to enforce network policies, perform load-balancing
+and provide encryption.
 
 .. include:: cni-chaining-limitations.rst
 
 .. important::
 
-   Due to a bug in certain version of the AWS CNI, please ensure that you are
-   running the AWS CNI `1.7.9 <https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.7.9>`_
-   or newer to guarantee compatibility with Cilium.
+   Please ensure that you are running version `1.7.9 <https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.7.9>`_
+   or newer of the AWS VPC CNI plugin to guarantee compatibility with Cilium.
+   The official upgrade instructions can be found `here <https://docs.aws.amazon.com/eks/latest/userguide/cni-upgrades.html>`_.
 
 .. image:: aws-cni-architecture.png
 
 
-Setup Cluster on AWS
-====================
+Setting up a cluster on AWS
+===========================
 
 Follow the instructions in the :ref:`k8s_install_quick` guide to set up an EKS
-cluster or use any other method of your preference to set up a Kubernetes
-cluster.
+cluster, or use any other method of your preference to set up a Kubernetes
+cluster on AWS.
 
-Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`__
-plugin is installed. If you have set up an EKS cluster, this is automatically
-done.
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed — which will already be the case if you have created an EKS
+cluster. Also, ensure the version of the plugin is up-to-date as per the above.
 
 .. include:: k8s-install-download-release.rst
 
-Deploy Cilium release via Helm:
+Deploy Cilium via Helm:
 
 .. parsed-literal::
 
    helm install cilium |CHART_RELEASE| \\
      --namespace kube-system \\
      --set cni.chainingMode=aws-cni \\
-     --set enableIPv4Masquerade=false \\
+     --set masquerade=false \\
```

Comment:
I don't think this option has any effect: it should either be `bpf.masquerade` if you want BPF masquerading, or `enableIPv4Masquerade` / `enableIPv6Masquerade` as beforehand.

Reply:
Nice one — I originally tested this with v1.9.x where `masquerade` is the right one. This also means that if we backport this we'll have to keep that in mind.

Reply:
Updated!